core/src/navigation/LeftNav.html

```diff
@@ -227,20 +376,78 @@
       },
 
       setExpandedState(nodes, value) {
+        if (this.getIsSemiCollapsed()) {
+          return;
+        }
+
         this.set({
           expandedCategories: NavigationHelpers.storeExpandedState(
             nodes.metaInfo.categoryUid,
             value
           )
         });
+      },
+
+      getIsSemiCollapsed() {
+        return localStorage.getItem(NavigationHelpers.COL_NAV_KEY) === 'true';
+      },
+
+      setIsSemiCollapsed(state) {
+        document.body.classList.remove('semiCollapsed');
+        // add if true
+        if (state) {
+          document.body.classList.add('semiCollapsed');
+        }
+        // initial state
+        this.set({ isSemiCollapsed: state });
+        localStorage.setItem(NavigationHelpers.COL_NAV_KEY, state);
+      },
+
+      closePopupMenu() {
+        if (this.get().selectedCategory) {
+          this.set({ selectedCategory: null });
+          document
+            .getElementsByClassName('fd-app__sidebar')[0]
+            .classList.remove('isBlocked');
+        }
+      },
+
+      semiCollapsibleButtonClicked(el) {
+        this.closePopupMenu();
+
+        if (!this.getIsSemiCollapsed()) {
+          this.setIsSemiCollapsed(true);
+
+          document
+            .getElementsByClassName('fd-side-nav')[0]
+            .setAttribute('style', 'overflow-y:hidden;');
+          window.setTimeout(function() {
+            document
+              .getElementsByClassName('fd-side-nav')[0]
+              .setAttribute('style', 'overflow-y:auto;');
+          }, 50);
```

Comment:
Why 50? If you want to have in the next tick, you can just leave the number empty.

Reply:
done